src/pages/index.js

```diff
@@ -0,0 +1,353 @@
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+  const likeButton = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  likeButton.addEventListener("click", (evt) => {
+    console.log("Se hizo click en el botón de like");
+    handleLike(evt, data._id);
+  });
+
+  const isLiked =
+    Array.isArray(data.likes) &&
+    data.likes.some((user) =>
+      typeof user === "string"
+        ? user === currentUserId
+        : user._id === currentUserId
+    );
+
+  if (data.isLiked) {
+    likeButton.classList.add("card__like-btn_liked");
+  }
+  function handleLike(evt, id) {
+    evt.preventDefault();
+    const likeButton = evt.target;
+
+    const isLiked = likeButton.classList.contains("card__like-btn_liked");
+
+    api
+      .changeLikeStatus(id, isLiked)
+      .then(() => {
+        likeButton.classList.toggle("card__like-btn_liked");
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  cardDeleteBtn.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  cardImageElement.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageElement.src = data.link;
+    previewModalImageElement.alt = data.name;
+    previewModalCaptionElement.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalEsc);
+}
+
+const modals = document.querySelectorAll(".modal");
+modals.forEach((modal) => {
+  modal.addEventListener("click", closeModalOverlay);
+});
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalEsc);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      //todo - call setButtonText instead
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+//todo implement loading ttext for all other form  submisiions
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Create", "Creating...");
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addNewCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset();
+      resetValidation(
```

Comment:
Please, delete it

`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submission. 

It’s a **waste of resources**.